pkg/kubelet/kuberuntime/kuberuntime_container_test.go

```diff
@@ -325,7 +325,7 @@ func TestLifeCycleHook(t *testing.T) {
 		}
 
 		// Now try to create a container, which should in turn invoke PostStart Hook
-		_, err := m.startContainer(fakeSandBox.Id, fakeSandBoxConfig, testContainer, testPod, fakePodStatus, nil, "", []string{})
+		_, err := m.startContainer(fakeSandBox.Id, fakeSandBoxConfig, testContainer, testPod, fakePodStatus, nil, "", []string{}, false)
```

Comment:
It concerns me that someone reading this code doesn't know what `false` means in the context of the `startContainer` method. At minimum, could we do `shouldRunAsync := false ... m.startContainer(... , shouldRunAsync)`. Perhaps even better could be defining a custom enum type?

My same thought applies to all the different places that `startContainer` is called.

Reply:
I agree with the readability suggestion.
If this approach receives more acceptance vote, I would refactor the code to make it more readable.

Reply:
@mattjmcnaughton 
I think if I implement #86017 in this PR, the boolean flag is not needed.
Whether post start hook is synchronous is determined by the type of container.

What do you think ?

Reply:
I have adopted your first suggestion by using a local variable shouldRunAsync.